actor-example/actors/src/temperature_monitor.rs

```diff
@@ -0,0 +1,63 @@
+use rtic_actor_traits::{Post, Receive};
+
+use crate::{TemperatureAlert, TemperatureReadingCelsius};
+
+pub struct TemperatureMonitor<P>
+where
+    P: Post<TemperatureAlert>,
+{
+    outbox: P,
+    threshold: i32,
+}
+
+impl<P> TemperatureMonitor<P>
+where
+    P: Post<TemperatureAlert>,
+{
+    pub fn new(outbox: P, threshold: i32) -> Self {
+        Self { outbox, threshold }
+    }
+}
+
+impl<P> Receive<TemperatureReadingCelsius> for TemperatureMonitor<P>
+where
+    P: Post<TemperatureAlert>,
+{
+    fn receive(&mut self, temperature: TemperatureReadingCelsius) {
+        if temperature.0 >= self.threshold {
+            self.outbox.post(TemperatureAlert).ok().expect("OOM");
```

Comment:
Why is `ok()` called here when you could call expect directly on the return from `post()` i.e.
`self.outbox.post(TemperatureAlert).expect("OOM");`

Reply:
doing `expect` on the `Result<T, E>` value requires that `E` implements the `fmt::Debug` trait. doing `ok().expect()` places no trait requirements on the type `E`. `TemperatureAlert` does not implement the `fmt::Debug` trait so `post(TemperatureAlert).expect()` would not compile.

deriving `Debug` on the type and leaving out the `ok()` is also an option (no pun intended). I tend to write `ok().expect()` mostly out of bad habit. (less `Result::unwrap` calls produces smaller (code size) binaries but that's micro-optimizing and this is an example so that hardly matters; as I said: bad habit)

Reply:
Ahh, ok makes sense. Thanks for the reply, I was mainly just curious more than critiquing